hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/utils/SparkMetadataWriterUtils.java

```diff
@@ -403,29 +403,30 @@ public static HoodiePairData<String, List<HoodieColumnRangeMetadata<Comparable>>
       // Step 2: Compute expression index records for the modified partitions
       LOG.debug("Indexing following columns for partition stats index: {}", validColumnsToIndex);
       List<String> partitionPaths = new ArrayList<>(commitMetadata.getWritePartitionPaths());
-      HoodieTableFileSystemView fileSystemView = getFileSystemView(dataMetaClient);
```

Comment:
My understanding is that we typically don't want to serialize the FSView. When using an ExternalSpillable FSView, this will fail since it is not serializable